src/librustc/mir/interpret/allocation.rs

```diff
@@ -187,6 +187,10 @@ impl<'tcx> rustc_serialize::UseSpecializedDecodable for &'tcx Allocation {}
 
 /// Byte accessors.
 impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
+    pub fn mem_pos(&self) -> MemoryPosition {
```

Comment:
remove the `size` and `align` fields and replace them with `MemoryPosition`

Reply:
I didn't do that earlier because for some reason `Allocation` derives `PartialOrd`, and merging the size and align would disrupt that.
Should I just remove the `PartialOrd` derive as well?

Reply:
Well.. `Allocation`'s `*Ord` is used by `ConstValue`'s `*Ord` which is used by `Const`'s `*Ord`, which is used by `UnpackedKind`'s `*Ord`, which is used by `TyKind`'s `*Ord`, which is used by `Kind`'s `*Ord`, which is used by ….

So, I guess not?